tests/test_backends.py

```diff
@@ -47,33 +48,76 @@ def test_get_backend_auto_detection_success_gguf(
     m_is_model_gguf.assert_called_once_with(tmp_gguf)
 
 
-# this test fails because the model_path is a directory but the platform is not linux
-@patch("sys.platform", "darwin")
-def test_get_backend_auto_detection_failure_vllm_dir_but_not_linux(
-    tmp_path: pathlib.Path,
-):
-    with pytest.raises(ValueError) as exc_info:
-        backends.get(tmp_path, None)
-    assert "Cannot determine which backend to use" in str(exc_info.value)
+# this tests both cases where a valid and invalid safetensors model directory is supplied
+@pytest.mark.parametrize(
+    "safetensors_model_path,expected",
+    [
+        (
+            pathlib.Path(
+                os.path.join(
+                    pathlib.Path(__file__).parent.resolve(),
+                    "testdata/valid_safetensors_model_dir",
+                )
+            ),
+            True,
+        ),
+        (
+            pathlib.Path(
+                os.path.join(
+                    pathlib.Path(__file__).parent.resolve(),
+                    "testdata/invalid_safetensors_model_dir",
+                )
+            ),
+            False,
+        ),
+    ],
+)
+def test_is_model_safetensors_valid(safetensors_model_path, expected):
```

Comment:
Change the parameters to "safetensor_dir,expected", [("valid_safetensors_model_dir", True), ...] and then use the `testdata_path` fixture to get the path.
Also add type hints to all parameters

```
def test_is_model_safetensors_valid(safetensors_dir: str, expected: bool, testdata_path: pathlib.Path) ->:
    safetensors_model_path = testdata_path / safetensors_dir
```

Reply:
updated, thanks!